docs/content/resource-authors-guide.md

```diff
@@ -94,12 +101,108 @@ presented to the user. You can also use the common Go patterm of returning `nil,
 err` in your `Check` and `Apply` statements. Converge will call `SetError(err)`
 automatically in this case. This gives you two options:
 
-1. call `SetError` yourself.
-1. return an error, which will be handled for you.
+1. call `SetError` yourself
+1. return an error, which will be handled for you
 
 You shoud choose *one* of these options and do it consistently across as much of
 your code as possible.
 
+## Task
+
+The
+[`resource.Task`](https://godoc.org/github.com/asteris-llc/converge/resource#Task) interface
+is what you will implement to have converge run your `Check` and `Apply`
+methods.  The `export` and `re-export-as` tags in your `resource.Task`
+implementation are used to define lookup methods from within converge.
+
+Using the `Shell` module as an example we can see how tasks should be
+implemented:
+
+```go
+type Shell struct {
+    CmdGenerator   CommandExecutor
+    CheckStmt      string                 `export:"check"`
+    ApplyStmt      string                 `export:"apply"`
+    Dir            string                 `export:"dir"`
+    Env            []string               `export:"env"`
+    Status         *CommandResults        `re-export-as:"status"`
+    CheckStatus    *CommandResults        `export:"checkstatus"`
+    HealthStatus   *resource.HealthStatus `export:"healthstatus"`
```

Comment:
Check status was so that the user would be able to access the results of `check` even during `apply`.  Otherwise you might get non-determinism if the result `check` and `apply` have different output.